src/main/java/com/datasonnet/header/Header.java

```diff
@@ -0,0 +1,123 @@
+            headerCommentSection = headerCommentSection.replace(DATASONNET_HEADER, "").replace("*/","");
+
+            JavaPropsMapper mapper = new JavaPropsMapper();
+            JavaPropsSchema schema = new JavaPropsSchema() {
+                class HeaderSplitter extends JPropPathSplitter
+                {
+                    protected final char _pathSeparatorChar = '.';
+
+                    public HeaderSplitter()
+                    {
+                        super(true);
+                    }
+
+                    @Override
+                    public JPropNode splitAndAdd(JPropNode parent,
+                                                 String key, String value)
+                    {
+                        JPropNode curr = parent;
+                        String[] segments = key.split("(?<!" + Pattern.quote("\\") + ")" + Pattern.quote("" + _pathSeparatorChar));
+                        for (String segment : segments) {
+                            curr = _addSegment(curr, segment.replaceAll("\\\\", ""));
+                        }
+                        return curr.setValue(value);
+                    }
+                }
+
+                @Override
+                public JPropPathSplitter pathSplitter() {
+                    return new HeaderSplitter();
+                };
+            };
+
+            try {
+                Properties props = new Properties();
+                props.load(new StringReader(headerCommentSection));
+                Map propsMap = mapper.readPropertiesAs(props, schema, Map.class);
+                if (propsMap.containsKey(DATASONNET_VERSION)) {
+                    header.setVersion((String)propsMap.get(DATASONNET_VERSION));
+                }
+                header.dataFormatParameters = propsMap;
+            } catch (Exception e) {
+                throw new HeaderParseException("Error parsing DataSonnet Header: ", e);
+            }
+        }
+
+        return header;
+    }
+
+    public String getVersion() {
+        return version;
+    }
+
+    public void setVersion(String version) {
+        this.version = version;
+    }
+
+    public Map<String, Map<String, Object>> getDefaultParameters(String mimeType) {
+        Map mergedParams = new HashMap();
+
+        Map defaultParams = (Map)dataFormatParameters.getOrDefault(Header.DATAFORMAT_DEFAULT, Collections.emptyMap());
+        Map defMimeTypeParams = (Map)defaultParams.getOrDefault(mimeType, Collections.emptyMap());
+        mergedParams.putAll(defMimeTypeParams);
+
+        return mergedParams;
+    }
+
+    public Map<String, Map<String, Object>> getInputParameters(String name, String mimeType) {
+        Map mergedParams = getDefaultParameters(mimeType);
+
+        Map input = (Map)dataFormatParameters.getOrDefault(Header.DATASONNET_INPUT, Collections.emptyMap());
+        Map defaultInput = (Map)input.getOrDefault(Header.DATAFORMAT_DEFAULT, Collections.emptyMap());
```

Comment:
I don't see much worth in having a default for just inputs as well as an overall default, and it adds some potential confusion, but I'm not super against it either